components/core/src/clp/regex_utils/regex_translation_utils.cpp

```diff
@@ -0,0 +1,227 @@
+
+class TranslatorState {
+public:
+    // Regex translation pattern analysis states.
+    enum class RegexPatternState : uint8_t {
+        // The initial state, where characters have no special meanings and are treated literally.
+        NORMAL = 0,
+        // Encountered a period `.`. Expecting wildcard expression.
+        DOT,
+        // Encountered a dollar sign `$`, meaning the regex string has reached the end anchor.
+        END,
+    };
+
+    // Constructor
+    TranslatorState() = default;
+
+    // Getters
+    [[nodiscard]] auto get_state() const -> RegexPatternState { return m_state; }
+
+    // Setters
+    auto set_next_state(RegexPatternState const& state) -> void { m_state = state; }
+
+private:
+    // Variables
+    RegexPatternState m_state{RegexPatternState::NORMAL};
+};
+}  // namespace
+
+namespace clp::regex_utils {
+// Main API
+auto regex_to_wildcard(string_view regex_str) -> BOOST_OUTCOME_V2_NAMESPACE::std_result<string> {
+    RegexToWildcardTranslatorConfig const default_config{false, false};
+    return regex_to_wildcard(regex_str, default_config);
+}
+
+auto regex_to_wildcard(string_view regex_str, RegexToWildcardTranslatorConfig const& config)
+        -> BOOST_OUTCOME_V2_NAMESPACE::std_result<string> {
+    if (regex_str.empty()) {
+        return string();
+    }
+
+    // Initialize translation state, scan position, and return string
+    TranslatorState state;
+    string_view::const_iterator it = regex_str.cbegin();
+    string wildcard_str;
+
+    // If there is no starting anchor character, append multichar wildcard prefix
+    if (cRegexStartAnchor == *it) {
+        ++it;
+    } else if (config.add_prefix_suffix_wildcards()) {
+        wildcard_str += cZeroOrMoreCharsWildcard;
+    }
+
+    error_code ec{};
+    while (it != regex_str.end()) {
+        // Main state transition table
+        switch (state.get_state()) {
+            case TranslatorState::RegexPatternState::NORMAL:
+                ec = normal_state_transition(state, it, wildcard_str, config);
+                break;
+            case TranslatorState::RegexPatternState::DOT:
+                ec = dot_state_transition(state, it, wildcard_str, config);
+                break;
+            case TranslatorState::RegexPatternState::END:
+                ec = end_state_transition(state, it, wildcard_str, config);
+                break;
+            default:
+                ec = ErrorCode::IllegalState;
+                break;
+        }
+        if (ec) {
+            return ec;
+        }
+        ++it;
+    }
+
+    // Do the final state check and clean up
+    ec = final_state_cleanup(state, it, wildcard_str, config);
+    if (ec) {
+        return ec;
+    }
+    return wildcard_str;
+}
+}  // namespace clp::regex_utils
+
+namespace {
+
+auto normal_state_transition(
+        TranslatorState& state,
+        string_view::const_iterator& it,
+        string& wildcard_str,
+        RegexToWildcardTranslatorConfig const& /*config*/
```

Comment:
```suggestion
        [[maybe_unused]] RegexToWildcardTranslatorConfig const& config
```
Do the same for the rest of methods that don't use `config`